internal/ais/parse.go

```diff
@@ -0,0 +1,128 @@
+package ais
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"os"
+)
+
+const ParseActivityName = "parse-mets-file"
+
+type (
+	ParseActivityParams struct {
+		METSPath string
+	}
+	ParseActivityResult struct {
+		MdRelPath   string
+		UAMdRelPath string
+	}
+	ParseActivity struct{}
+)
+
+func NewParseActivity() *ParseActivity {
+	return &ParseActivity{}
+}
+
+// Execute parses the METS file at params.METSPath and returns the relative
+// paths to the AIP data folder for the UpdatedAreldaMetadata.xml and
+// metadata.xml files.
+func (a *ParseActivity) Execute(ctx context.Context, params *ParseActivityParams) (*ParseActivityResult, error) {
+	mets, err := os.Open(params.METSPath)
+	if err != nil {
+		return nil, fmt.Errorf("ParseActivity: open METS: %w", err)
+	}
+	defer mets.Close()
+
+	var (
+		inFileGrp, inStructMap, inMdDiv, inUAMdDiv bool
+		currentFileID, mdFileID, uamdFileID        string
+	)
+
+	// Stream the METS file creating a map of metadata file IDs to their
+	// relative paths from the fileSec section. Obtain the file IDs for
+	// the UpdatedAreldaMetadata.xml and metadata.xml files from the
+	// physical structMap section.
+	mdFiles := make(map[string]string)
+	decoder := xml.NewDecoder(mets)
+	for {
+		tok, err := decoder.Token()
+		if err != nil {
+			if err.Error() == "EOF" {
+				break
+			}
+			return nil, fmt.Errorf("ParseActivity: parse METS: %w", err)
+		}
+
+		switch elem := tok.(type) {
+		case xml.StartElement:
+			switch elem.Name.Local {
+			case "fileGrp":
+				if getAttr(elem, "USE") == "metadata" {
+					inFileGrp = true
+				}
+			case "file":
+				if inFileGrp {
+					currentFileID = getAttr(elem, "ID")
+				}
+			case "FLocat":
+				if currentFileID != "" {
+					mdFiles[currentFileID] = getAttr(elem, "href")
+				}
+			case "structMap":
+				if getAttr(elem, "TYPE") == "physical" {
+					inStructMap = true
+				}
```

Comment:
Also add a `inHeadDir` or something like that with an extra check to be "sure" we are not picking another metadata.xml file.